src/ProjectOrigin.VerifiableEventStore/Services/Batcher/MemoryBatcher.cs

```diff
@@ -0,0 +1,57 @@
+using Microsoft.Extensions.Options;
+using ProjectOrigin.VerifiableEventStore.Extensions;
+using ProjectOrigin.VerifiableEventStore.Models;
+using ProjectOrigin.VerifiableEventStore.Services.BlockchainConnector;
+using ProjectOrigin.VerifiableEventStore.Services.EventStore;
+
+namespace ProjectOrigin.VerifiableEventStore.Services.Batcher;
+
+public class MemoryBatcher : IBatcher
+{
+    private IBlockchainConnector blockchainConnector;
+    private IEventStore eventStore;
+    private IOptions<BatcherOptions> options;
+    private List<Event> events = new List<Event>();
+    private long batchSize;
+    private const int blockRetryWaitMilliseconds = 1000;
+
+    public MemoryBatcher(IBlockchainConnector blockchainConnector, IEventStore eventStore, IOptions<BatcherOptions> options)
+    {
+        this.blockchainConnector = blockchainConnector;
+        this.eventStore = eventStore;
+        this.options = options;
+
+        batchSize = (long)Math.Pow(2, options.Value.BatchSizeExponent);
+    }
+
+    public async Task PublishEvent(Event e)
```

Comment:
Does the end-user get a ticket or sub-proof or anything upon batching the whole set of events ? Or is this on the to-do list for our beloved production team :-)